src/main/java/org/springframework/integration/kafka/listener/ConcurrentMessageListenerDispatcher.java

```diff
@@ -100,6 +104,8 @@ public void stop(int stopTimeout) {
 				this.running = false;
 				delegates.flip().keyBag().toSet().forEachWith(stopDelegateProcedure, stopTimeout);
 			}
+			this.taskExecutor = null;
```

Comment:
Need a setter for the TE in case they're running in a JEE container; also we can't null it if set by the user.